prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala

```diff
@@ -745,14 +768,14 @@ class ParserSpec extends AnyFunSpec with Matchers {
   }
 
   private def parseSuccessfully(query: String) = {
-    val result = LegacyParser.parseQuery(query)
+    val result = LegacyParser.parseQuery(query).requireMetricNames()
```

Comment:
You'd need to be careful to call `requireMetricNames` only when `LegacyParser` and `AntlrParser` methods are directly accessed. Otherwise, all public methods of `Parser` (except `parseQueryWithoutMetricNameRequirement`) will call it automatically.

Reply:
The general idea here is that `Parser` enforces `filodb` constraints when it wants to, but the other two parsers don't.